warehouse/query-core/src/main/java/datawave/query/util/DateIndexHelper.java

```diff
@@ -170,39 +173,93 @@ public DateIndexHelper initialize(AccumuloClient client, String dateIndexTableNa
+         *
+         * @param startDate
+         *            the start date
+         * @param endDate
+         *            the end date
+         */
+        public void setStartEndDate(Date startDate, Date endDate) {
+            if (start == null) {
+                start = DateIndexUtil.format(startDate);
+            }
+            if (end == null) {
+                end = DateIndexUtil.format(endDate);
+            }
+        }
+
+        /**
+         * Method that updates the start and end date if the provided date is beyond either bound
+         *
+         * @param date
+         *            the date
+         */
+        public void updateStartEndDate(String date) {
+            if (start == null || date.compareTo(start) < 0) {
+                start = date;
+            }
+            if (end == null || date.compareTo(end) > 0) {
+                end = date;
+            }
         }
 
         public Date getBeginDate() {
             try {
-                return DateIndexUtil.getBeginDate(dateRange[0]);
+                return DateIndexUtil.getBeginDate(start);
             } catch (ParseException pe) {
-                log.error("Malformed date in date index table, expected yyyyMMdd: " + dateRange[0], pe);
-                throw new IllegalStateException("Malformed date in date index table, expected yyyyMMdd: " + dateRange[0], pe);
+                log.error("Malformed date in date index table, expected yyyyMMdd: " + start, pe);
+                throw new IllegalStateException("Malformed date in date index table, expected yyyyMMdd: " + start, pe);
             }
         }
 
         public Date getEndDate() {
             try {
-                return DateIndexUtil.getEndDate(dateRange[1]);
+                return DateIndexUtil.getEndDate(end);
             } catch (ParseException pe) {
-                log.error("Malformed date in date index table, expected yyyyMMdd: " + dateRange[1], pe);
-                throw new IllegalStateException("Malformed date in date index table, expected yyyyMMdd: " + dateRange[1], pe);
+                log.error("Malformed date in date index table, expected yyyyMMdd: " + end, pe);
+                throw new IllegalStateException("Malformed date in date index table, expected yyyyMMdd: " + end, pe);
             }
         }
 
         public String toString() {
-            StringBuilder builder = new StringBuilder();
-            builder.append("field: ").append(fields);
-            builder.append(", dateRange").append(Arrays.asList(dateRange));
-            return builder.toString();
+            return "field: " + fields + ", dateRange[" + start + "," + end + "]";
+        }
+
+        public String serializeToString() {
+            StringBuilder sb = new StringBuilder();
+            sb.append(start).append(',');
+            sb.append(end).append(',');
+            sb.append(Joiner.on(',').join(fields));
```

Comment:
Joiner instances are thread-safe and immutable, so you could have a static final constant for reuse throughout the class.